applications/luci-app-attendedsysupgrade/htdocs/luci-static/resources/view/attendedsysupgrade/overview.js

```diff
@@ -9,412 +9,530 @@
-                                else
-                                        displayStatus("notice spinning", E('p', _('Building firmware sysupgrade image')));
-
-                                setTimeout(function() {
-                                        request_sysupgrade(server_url, data);
-                                }, 5000);
-                                break;
-                        case 400: // bad request
-                        case 422: // bad package
-                        case 500: // build failed
-                                res = response.json()
-                                var body = [
-                                        E('p', {}, res.detail),
-                                        E('p', {}, _("Please report the error message and request")),
-                                        E('b', {}, _("Request to server:")),
-                                        E('pre', {}, JSON.stringify(data, null, 4)),
-
-                                ]
-
-                                if (res.stdout) {
-                                        body.push(E('b', {}, "STDOUT:"))
-                                        body.push(E('pre', {}, res.stdout))
-
-                                }
-
-                                if (res.stderr) {
-                                        body.push(E('b', {}, "STDERR:"))
-                                        body.push(E('pre', {}, res.stderr))
-
-                                }
-
-                                body = body.concat([
-                                        E('div', {
-                                                'class': 'right'
-                                        }, [
-                                                E('div', {
-                                                        'class': 'btn',
-                                                        'click': ui.hideModal
-                                                }, _('Close'))
-                                        ])
-                                ]);
-                                ui.showModal(_('Error building the sysupgrade'), body);
-                                break;
-                }
-        });
+	var res, req;
+
+	if (data.request_hash) {
+		req = request.get(`${server_url}/api/v1/build/${data.request_hash}`);
+	} else {
+		req = request.post(`${server_url}/api/v1/build`, {
+			profile: data.board_name,
+			target: data.target,
+			version: data.version,
+			packages: data.packages,
+			diff_packages: true,
+		});
+	}
+
+	req.then((response) => {
+		switch (response.status) {
+			case 200:
+				res = response.json();
+				var image;
+				for (image of res.images) {
+					if (image.type == 'sysupgrade') {
+						break;
+					}
+				}
+				if (image.name != undefined) {
+					var sysupgrade_url = `${server_url}/store/${res.bin_dir}/${image.name}`;
```

Comment:
I was just about to merge and went back over the pullrequest and saw the wrong comma again. Sorry
I would suggest that we use `'` over all.
If you are building a string then I would always use the `format` function.